src/jit/flowgraph.cpp

```diff
@@ -5843,28 +5866,14 @@ void Compiler::fgFindBasicBlocks()
 
     /* Now create the basic blocks */
 
-    fgMakeBasicBlocks(info.compCode, info.compILCodeSize, jumpTarget);
+    unsigned retBlocks = fgMakeBasicBlocks(info.compCode, info.compILCodeSize, jumpTarget);
 
     if (compIsForInlining())
     {
-        bool hasReturnBlocks           = false;
-        bool hasMoreThanOneReturnBlock = false;
-
-        for (BasicBlock* block = fgFirstBB; block != nullptr; block = block->bbNext)
-        {
-            if (block->bbJumpKind == BBJ_RETURN)
-            {
-                if (hasReturnBlocks)
-                {
-                    hasMoreThanOneReturnBlock = true;
-                    break;
-                }
-
-                hasReturnBlocks = true;
-            }
-        }
-
-        if (!hasReturnBlocks && !compInlineResult->UsesLegacyPolicy())
+		// If fgFindJumpTargets marked this as "no return"  there really should be no BBJ_RETURN blocks in the method
+		noway_assert(retBlocks > 0 && ((impInlineInfo->iciCall->gtCallMoreFlags & GTF_CALL_M_DOES_NOT_RETURN) != 0));
```

Comment:
Well, I'm not sure which is before and which is after. The latest commit still uses `retBlocks > 0 && ...` which is obviously incorrect.

Since the language doesn't offer a logical implication operator you have to emulate it - `A -> B` is implemented as `!A || B`. That is, "the flag is not set or the ret block count is 0".